x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/action/TransportPutRollupJobAction.java

```diff
@@ -124,6 +132,40 @@ private static RollupJob createRollupJob(RollupJobConfig config, ThreadPool thre
         return new RollupJob(config, filteredHeaders);
     }
 
+    static RollupJobConfig addDefaultMetricsIfNecessary(final RollupJobConfig originalConfig, final ClusterState clusterState) {
+        // All nodes need to support `date` field types in RollupJobConfig#metricConfig settings
+        if (clusterState.nodes().getMinNodeVersion().before(Version.V_6_5_0)) {
+            return originalConfig;
+        }
+        final GroupConfig groupConfig = originalConfig.getGroupConfig();
+
+        List<MetricConfig> inputMetrics =
+            originalConfig.getMetricsConfig() != null ? new ArrayList<>(originalConfig.getMetricsConfig()) : new ArrayList<>();
+        if (groupConfig != null) {
+            String timeField = groupConfig.getDateHistogram().getField();
+            Set<String> currentFields = inputMetrics.stream().map(MetricConfig::getField).collect(Collectors.toSet());
+            if (currentFields.contains(timeField) == false) {
```

Comment:
Should we also handle the case where metrics are defined for the time field but `min` or `max` are missing ?

Reply:
I don't think so. The idea here is that if the user specified metrics for the timefield they want those specific metrics only and have put thought into why, at least that is the thought process when I discussed with @colings86 and @polyfractal

Reply:
What I want to avoid is the situation where the user wants to limit the metrics to save space and doesn't need ML either. In this situation it would be annoying if we force them to have these metrics just because we like having them. If they did try to use ML they should just get a polite notice saying that because they chose not to store these metrics they can't create a job. Also we should make sure the documentation for ML and for rollups both point out this. The thing we have solved (and what we set out to solve) is the default case for rollups being able to support ML jobs. Happy to discuss this though if others have a different view.

Reply:
I understant the intent but what is the expected way to opt out from these metrics. Users have to set an empty list of metrics for the timestamp field ? I don't think that's allowed currently so they'll need to pick one metric to make sure that we don't add two ? IMHO I'd prefer a clear option to opt out from this behavior.

Reply:
@jimczi @colings86 let me know what you think of the latest commit. I have allowed empty metrics to be provided for a metric config. 

I added integration tests and some docs about it as well. All seems well from what I can tell without any changes to the internal logic of Rollup.

Reply:
I'd prefer an additional request param when creating the job. Silently ignoring empty list of metrics seems dangerous IMHO.

Reply:
I am not sure how it is dangerous? The end user still has to explicitly supply a metric object, and the passed array cannot be null. Same goes for the API payload, if a metric object is supplied but no metrics given, an error will still return. 
Also, if all metrics objects just contain empty metrics, an error is still returned to the user. 

All that said, this is not my wheelhouse, so I will happily change it to a request parameter.